### PR TITLE
OPSEXP-2277 Migration of APS helm chart

### PR DESCRIPTION
**Pull request by pmacius**
Ref: OPSEXP-2277

### CODE REVIEW

File: .github/workflows/lint-test.yaml [hunk 1]
```diff
@@ -67,12 +67,21 @@ jobs:
...
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_S3_ACSLICENSE_SECRET_ACCESS_KEY }}
+        run: aws s3 cp s3://aps-licenses/single-tenant/activiti.lic /tmp/activiti.lic
+
       - name: Prepare namespace for install
         if: steps.list-changed.outputs.changed == 'true'
         run: |
           kubectl create ns "$INSTALL_NAMESPACE"
           kubectl create secret generic quay-registry-secret --from-file=.dockerconfigjson="${HOME}"/.docker/config.json --type=kubernetes.io/dockerconfigjson -n "$INSTALL_NAMESPACE"
           kubectl create secret generic broker-secret --from-literal=BROKER_URL="failover:(nio://activemq:61616)?timeout=3000" --from-literal=BROKER_USERNAME=admin --from-literal=BROKER_PASSWORD=admin -n "$INSTALL_NAMESPACE"
+          kubectl create secret generic --from-file=activiti.lic=/tmp/activiti.lic aps-license -n "$INSTALL_NAMESPACE"
```
Comment (gionn) [inline .github/workflows/lint-test.yaml:84]:
```suggestion
          kubectl create secret generic aps-license --from-file=activiti.lic=/tmp/activiti.lic -n "$INSTALL_NAMESPACE"
```

File: charts/alfresco-process-services/README.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+| processEngine.readinessProbe.timeoutSeconds | int | `5` |  |
+| processEngine.replicacount | int | `1` |  |
+| processEngine.service.externalPort | int | `80` |  |
+| processEngine.service.name | string | `"aps"` |  |
+| processEngine.service.type | string | `"ClusterIP"` |  |
+| registryPullSecrets | string | `"quay-registry-secret"` |  |
+| replicaCount | int | `1` |  |
+
+----------------------------------------------
+Autogenerated from chart metadata using [helm-docs v1.11.0](https://github.com/norwoodj/helm-docs/releases/v1.11.0)
```
Comment (gionn) [inline charts/alfresco-process-services/README.md:113]: copy the readme tpl from another chart

File: charts/alfresco-process-services/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+apiVersion: v2
+description: A Helm chart for Alfresco Process Services
+name: alfresco-process-services
+version: 0.11.0-alpha.0
+appVersion: 24.1.0
+maintainers:
+  - name: Alfresco
```
Comment (alxgomz) [inline charts/alfresco-process-services/Chart.yaml:7]: Please update to `Hyland` or remove it (looks like we're not setting this in others charts)

File: charts/alfresco-process-services/templates/config-admin-database.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+{{- if not .Values.database.existingConfigMap.name }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: >-
+    {{ template "alfresco-process-services.database-config-admin.name" . }}
+  labels:
+    {{- include "alfresco-process-services.labels" . | nindent 4 }}
```
Comment (alxgomz) [inline charts/alfresco-process-services/templates/config-admin-database.yaml:8]:
```suggestion
    {{- include "alfresco-process-services.admin.labels" . | nindent 4 }}
```

File: charts/alfresco-process-services/templates/config-admin.yaml [hunk 5]
```diff
@@ -0,0 +1,14 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  # the name of the config map
+  name: {{ template "alfresco-process-services.config-admin.name" . }}
+  labels:
+    {{- include "alfresco-process-services.labels" . | nindent 4 }}
```
Comment (alxgomz) [inline charts/alfresco-process-services/templates/config-admin.yaml:7]:
```suggestion
    {{- include "alfresco-process-services.admin.labels" . | nindent 4 }}
```

File: charts/alfresco-process-services/templates/config-aps-database.yaml [hunk 6]
```diff
@@ -0,0 +1,13 @@
+{{- if not .Values.database.existingConfigMap.name }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: >-
+    {{ template "alfresco-process-services.database-config-aps.name" . }}
+  labels:
+    {{- include "alfresco-process-services.labels" . | nindent 4 }}
```
Comment (alxgomz) [inline charts/alfresco-process-services/templates/config-aps-database.yaml:8]:
```suggestion
    {{- include "alfresco-process-services.aps.labels" . | nindent 4 }}
```

File: charts/alfresco-process-services/templates/deployment-admin.yaml [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    {{- include "alfresco-process-services.admin.labels" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      {{- include "alfresco-process-services.admin.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      annotations:
+        checksum/config: {{ include (print $.Template.BasePath "/config-admin.yaml") . | sha256sum }}
```
Comment (alxgomz) [inline charts/alfresco-process-services/templates/deployment-admin.yaml:15]: Please remove. This reload method is pretty hacky and rather unreliable.

File: charts/alfresco-process-services/templates/deployment-admin.yaml [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+                  key: {{ .existingSecret.keys.password }}
+            - name: ACTIVITI_ADMIN_DATASOURCE_URL
+              valueFrom:
+                configMapKeyRef:
+                  name: {{ $dbCm }}
+                  key: {{ .existingConfigMap.keys.url }}
+            {{- end }}
+          envFrom:
+          - configMapRef:
+              # config map to use, defined in config-repository.yaml
```
Comment (alxgomz) [inline charts/alfresco-process-services/templates/deployment-admin.yaml:47]: Looks like a n old copy/paste. there is no such file in this chart)

File: charts/alfresco-process-services/templates/ingress-admin.yaml [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+  name: {{ template "alfresco-process-services.ingress-admin.name" . }}
+  labels:
+    {{- include "alfresco-process-services.labels" . | nindent 4 }}
+  annotations:
+    ingress.kubernetes.io/ssl-redirect: "false"
+    {{- with .Values.adminApp.ingress.maxUploadSize }}
+    nginx.ingress.kubernetes.io/proxy-body-size: {{ . }}
+    {{- end }}
+spec:
+  ingressClassName: "nginx"
```
Comment (alxgomz) [inline charts/alfresco-process-services/templates/ingress-admin.yaml:16]: Can you make it a value please?

File: charts/alfresco-process-services/templates/ingress-aps.yaml
Comment (alxgomz) [inline charts/alfresco-process-services/templates/ingress-aps.yaml:1]: looks like an old ingress template. If that's too much for this PR we should create another task to migrate it to something more aligned with e.g. https://github.com/Alfresco/alfresco-helm-charts/blob/main/charts/alfresco-repository/templates/ingress.yaml

File: charts/alfresco-process-services/templates/svc-admin.yaml [hunk 10]
```diff
@@ -0,0 +1,14 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ template "alfresco-process-services.service-admin.name" . }}
+  labels:
+    {{- include "alfresco-process-services.labels" . | nindent 4 }}
```
Comment (alxgomz) [inline charts/alfresco-process-services/templates/svc-admin.yaml:6]:
```suggestion
    {{- include "alfresco-process-services.admin.labels" . | nindent 4 }}
```

File: charts/alfresco-process-services/values.yaml [hunk 11]
```diff
@@ -0,0 +1,185 @@
+# Default values for Alfresco Process Services.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+---
+replicaCount: 1
```
Comment (alxgomz) [inline charts/alfresco-process-services/values.yaml:5]: I would move it under `.adminApp` for the sake of consistency

File: charts/alfresco-process-services/values.yaml [hunk 12]
```diff
@@ -0,0 +1,185 @@
...
+  service:
+    name: aps
+    type: ClusterIP
+    externalPort: 80
+  ingress:
+    path: /activiti-app
+    maxUploadSize: "5G"
+  persistence:
+    # -- Persist processEngine data
+    enabled: true
```
Comment (alxgomz) [inline charts/alfresco-process-services/values.yaml:34]: other component charts have it set to false b default.

File: charts/alfresco-process-services/values.yaml [hunk 13]
```diff
@@ -0,0 +1,185 @@
...
+
+database:
+  # -- JDBC url to connect to the external DB
+  url:
+    aps: null
+    admin: null
+  # -- JDBC username to use to connect to the DB
+  username: null
+  # -- JDBC password to use to connect to the DB
+  password: null
```
Comment (alxgomz) [inline charts/alfresco-process-services/values.yaml:168]: Ideally though could be different for aps and admin app (could also be another task)

File: charts/alfresco-process-services/values.yaml [hunk 14]
```diff
@@ -0,0 +1,185 @@
...
+  existingConfigMap:
+    name: null
+    keys:
+      # -- configmap key where to find the URL of the database
+      url: DATABASE_URL
+      # -- configmap key where to find the JDBC driver class to use.
+      # The configmap may leverage the alfresco-repository.db.cm named
+      # template to auto-generate it from the sole url parameter.
+      driver: DATABASE_DRIVER
```
Comment (alxgomz) [inline charts/alfresco-process-services/values.yaml:177]: look like this is not used and instead we're using the `cm` projected in deployment with `envFrom`

File: charts/alfresco-process-services/values.yaml [hunk 15]
```diff
@@ -0,0 +1,185 @@
...
+  username: null
+  # -- JDBC password to use to connect to the DB
+  password: null
+  existingConfigMap:
+    name: null
+    keys:
+      # -- configmap key where to find the URL of the database
+      url: DATABASE_URL
+      # -- configmap key where to find the JDBC driver class to use.
+      # The configmap may leverage the alfresco-repository.db.cm named
```
Comment (alxgomz) [inline charts/alfresco-process-services/values.yaml:175]: `alfresco-repository.db.cm` is not available here unless àlfresco-repository` chart is also a dependency. Please remove.

**Review by alxgomz — CHANGES_REQUESTED** [review]
I think we need a major bump as the configuration i already different.
Also please add `podLabels`, `podAnnotations` , `taints` & `tolerations`

Current `.database.existingConfigMap` do not allow setting different username/pass/url for adminpp and aps which is needed. We may need somethign like `.database.[aps|admin].*`

File: charts/alfresco-process-services/values.yaml (hunk 13 above)
Comment (gionn) [reply charts/alfresco-process-services/values.yaml:168]: +1 for handling existingConfigMap/Secret in a new task/pr

File: charts/alfresco-process-services/values.yaml [hunk 16]
```diff
@@ -0,0 +1,185 @@
...
+    # -- activiti-app password
+    ACTIVITI_ADMIN_REST_APP_PASSWORD: "admin"
+  volumes: []
+  volumeMounts: []
+
+database:
+  # -- JDBC url to connect to the external DB
+  url:
+    aps: null
+    admin: null
```
Comment (gionn) [inline charts/alfresco-process-services/values.yaml:164]: in general I think it's better to split the other way around (e.g. processEngine.database.url and adminApp.database.url)

**Review by gionn — CHANGES_REQUESTED** [review]

**Review by gionn — APPROVED** [review]

**Review by alxgomz — APPROVED** [review]

**Pull request merged**